### PR TITLE
Event Creation

### PR DESCRIPTION
**Pull request by AbhinavGupta2002**
Implemented the feature where club admins can add new events for their respective clubs.

Things to note:

1. POST `api/v1/club/:id/event` API was updated to support accepting `private_flag` and `location` fields.
2. For club admins, the ability to make an event private will appear if and only if the club has a membership fee.
3. The view, view model, and view model factory added in the frontend for the PR is for the `AddEvent` component.

### CODE REVIEW

File: app/app/src/main/java/com/example/clubwat/viewmodels/AddEventViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            var isEventAdded = false
+            try {
+                val url = URL(String.format(BuildConfig.ADD_EVENT_URL, clubId))
+                (url.openConnection() as HttpURLConnection).apply {
+                    requestMethod = "POST"
+                    doOutput = true
+                    setRequestProperty("Content-Type", "application/json")
+                    setRequestProperty("Authorization", "Bearer " + userRepository.currentUser.value!!.userId )
+
+                    val body = ("""{"title": "${title.value.trim()}",""" +
```
Comment (anjalig21) [inline app/app/src/main/java/com/example/clubwat/viewmodels/AddEventViewModel.kt:70]: I think we can make this cleaner by using JSONObject and then toString as done in the codebase already

Comment (AbhinavGupta2002) [reply app/app/src/main/java/com/example/clubwat/viewmodels/AddEventViewModel.kt]: resolved.

File: app/app/src/main/java/com/example/clubwat/views/ClubDetailsView.kt [hunk 2]
```diff
@@ -189,6 +188,11 @@ fun ClubDetailsView(
                     )
                     if (club != null) {
                         LazyColumn {
+                            if (club!!.isClubAdmin) {
```
Comment (Kggupta) [inline app/app/src/main/java/com/example/clubwat/views/ClubDetailsView.kt:191]: This should be in club management view. Copy what was done with user management

File: app/app/src/main/java/com/example/clubwat/views/AddEventView.kt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                    DateTimePickerView(dateTime = viewModel.startDateTime)
+                }
+
+                Column() {
+                    Text(text = "End Date", style = TextStyle(fontWeight = FontWeight.Bold))
+                    DateTimePickerView(dateTime = viewModel.endDateTime)
+                }
+
+                Column() {
+                    Text(text = "Location", style = TextStyle(fontWeight = FontWeight.Bold))
```
Comment (Kggupta) [inline app/app/src/main/java/com/example/clubwat/views/AddEventView.kt:121]: This should be a text field not a radio button

File: app/app/src/main/java/com/example/clubwat/views/AddEventView.kt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                            focusedIndicatorColor = Orange,
+                            unfocusedContainerColor = Color.Transparent,
+                            focusedContainerColor = Color.Transparent
+                        ),
+                        singleLine = true
+                    )
+                }
+
+                Column() {
+                    Text(text = "Description", style = TextStyle(fontWeight = FontWeight.Bold))
```
Comment (Kggupta) [inline app/app/src/main/java/com/example/clubwat/views/AddEventView.kt:94]: This should be a multiline text box

**Review by Kggupta — CHANGES_REQUESTED** [review]
see comments

File: app/app/src/main/java/com/example/clubwat/views/AddEventView.kt [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+                            }
+                        )
+                        Text(
+                            text = "In-person",
+                            modifier = Modifier.align(Alignment.CenterVertically)
+                        )
+                    }
+                }
+
+                if (isClubPaid == true) {
```
Comment (Kggupta) [inline app/app/src/main/java/com/example/clubwat/views/AddEventView.kt:151]: we don't need this restriction

File: app/app/src/main/java/com/example/clubwat/views/ClubDetailsView.kt (hunk 2 above)
Comment (AbhinavGupta2002) [reply app/app/src/main/java/com/example/clubwat/views/ClubDetailsView.kt]: resolved.

File: app/app/src/main/java/com/example/clubwat/views/AddEventView.kt (hunk 5 above)
Comment (AbhinavGupta2002) [reply app/app/src/main/java/com/example/clubwat/views/AddEventView.kt]: resolved.

File: app/app/src/main/java/com/example/clubwat/views/AddEventView.kt (hunk 3 above)
Comment (AbhinavGupta2002) [reply app/app/src/main/java/com/example/clubwat/views/AddEventView.kt:123]: resolved.

**Review by Kggupta — APPROVED** [review]

**Pull request merged**